subworkflows/local/samplesheet_creation.nf

```diff
@@ -0,0 +1,115 @@
+include { CAT_CAT               } from '../../modules/nf-core/cat/cat/main'
+include { MAG_TO_SAMPLESHEET    } from '../../modules/local/mag_to_samplesheet'
+include { MAG_MERGE_SAMPLESHEET } from '../../modules/local/mag_merge_samplesheet'
+
+workflow SAMPLESHEET_CREATION {
+    take:
+        short_reads //channel: [val(meta), path(fastq_1), path(fastq_2)]
+        assemblies  //channel: [val(meta), path(fasta)]
+    main:
+        ch_versions = Channel.empty()
+
+        // combine assemblies by sample/group if multiple assembly methods were used
+        ch_assemblies = assemblies
+            .map {
+                meta, fasta ->
+                    def meta_new = meta.subMap('id')
+                [ meta_new, fasta ]
+            }
+            .groupTuple()
+
+        //
+        // MODULE: Combine all assemblies from a sample into one FastA file
+        //
+        ch_combined_assemblies = CAT_CAT ( ch_assemblies ).file_out
+        ch_versions = ch_versions.mix( CAT_CAT.out.versions )
```

Comment:
I guess this maybe would not apply to all pipelines? I wonder if you should move much of this into a pipeline specific creation subworkflow, so this file would have 

```
if { nf_core_pipeline == phage_annotator } {
   CREEATE_ PHAGEANNOTATOR_SAMPLESHEET (xyz)
} (nf_core_pipeline == funcscan {
  CRERATE_FUNCSCAN_SAMPLESHEET
}
```

This would then make it more flexible for other pipelines